src/WebJobs.Script.Grpc/MessageExtensions/StatusResultExtensions.cs

```diff
@@ -43,8 +43,8 @@ public static bool IsInvocationSuccess<T>(this StatusResult status, TaskCompleti
                     return false;
 
                 case StatusResult.Types.Status.Cancelled:
-                    tcs.SetCanceled();
-                    return false;
+                    var cancellationException = GetRpcException(status);
+                    throw new OperationCanceledException("Invocation was cancelled", cancellationException);
```

Comment:
Just want to confirm, we definitely want to throw here?

todo item for me: validate what happens in-proc